gemrb/core/DataFileMgr.h

```diff
@@ -38,13 +42,127 @@ namespace GemRB {
  * Abstract loader for .INI files
  */
 
+class KeyValueGroup {
+public:
+	using KeyValueMap = StringMap<std::string>;
+
+private:
+	KeyValueMap kvMap;
+	const std::string name;
+
+public:
+	KeyValueGroup (std::string name) : name(std::move(name)) {}
+
+	const std::string& GetName() const {
+		return name;
+	}
+
+	auto begin() const {
+		return kvMap.begin();
+	}
+
+	auto end() const {
+		return kvMap.end();
+	}
+
+	size_t size() const {
+		return kvMap.size();
+	}
+
+	template<typename T>
+	T GetAs(StringView key, const T) const = delete;
+
+	bool AddLine(StringView iniLine) {
+		auto equalsPos = FindFirstOf(iniLine, "=");
+		if (equalsPos == std::string::npos) {
+			return false;
+		}
+
+		auto keyStart = FindFirstNotOf(iniLine, " ");
+		if (keyStart == std::string::npos) {
+			return true;
+		}
+
+		StringView key = RTrimCopy(SubStr(iniLine, keyStart, equalsPos - keyStart));
+
+		auto valueStart = FindFirstNotOf(iniLine, " ", equalsPos + 1);
+		if (valueStart == std::string::npos) {
+			return true;
+		}
+
+		auto valueEnd = FindLastNotOf(iniLine, " ");
+		if (valueEnd == std::string::npos) {
+			return true;
+		}
+
+		std::string value = SubStr(iniLine, valueStart, valueEnd - valueStart + 1).MakeString();
+		kvMap.Set(key, std::move(value));
```

Comment:
you can instead do `StringView value` and `kvMap.Set(key, value.MakeString())`

Reply:
I just thought it was better readability

```
std::string value = SubStr(iniLine, valueStart, valueEnd - valueStart + 1).MakeString();
kvMap.Set(key, std::move(value));
```
vs
```
StringView value = SubStr(iniLine, valueStart, valueEnd - valueStart + 1);
kvMap.Set(key, value.MakeString());
```

Reply:
Ok. Thought I was missing something here.